lib/database/adapters/RunAdapter.js

```diff
@@ -239,7 +245,7 @@ class RunAdapter {
             pdpConfigOption: entityObject.pdpConfigOption,
             pdpTopologyDescriptionLibraryFile: entityObject.pdpTopologyDescriptionLibraryFile,
             tfbDdMode: entityObject.tfbDdMode,
-            lhcPeriod: entityObject.lhcPeriod,
+            lhcPeriod: { name: entityObject.lhcPeriod },
```

Comment:
If at this point entityObject.lhcPeriod is a string, it means it's not in a valid state.
In the entity (service layer, which is, or should be, the core layer), the `lhcPeriod` is already an object => call the appropriate adapter (check for null first)

Reply:
No. Here, `entityObject.lhcPeriod` must be an instance of `LhcPeriod` or null.